code/game/gamemodes/autotraitor/autotraitor.dm

```diff
@@ -57,11 +57,10 @@
 
 	for(var/i = 0, i < num_traitors, i++)
 		var/datum/mind/traitor = pick(possible_traitors)
+		possible_traitors -= traitor
 		if(traitor.special_role)
-			possible_traitors.Remove(traitor)
```

Comment:
This shouldn't be the issue, because of the continue immediately after, which is why we should be deleting continue and break from loops for being jump statements.